packages/transport/src/modules/sync/DatawalletModificationsProcessor.ts

```diff
@@ -61,41 +56,60 @@ export class DatawalletModificationsProcessor {
     ];
 
     public async execute(): Promise<void> {
-        await this.applyCreates();
-        await this.applyUpdates();
-        await this.applyDeletes();
+        await this.applyModifications();
         await this.applyCacheChanges();
     }
 
-    private async applyCreates() {
-        if (this.creates.length === 0) {
-            return;
-        }
-
-        const createsGroupedByObjectIdentifier = _.groupBy(this.creates, (c) => c.objectIdentifier);
+    private async applyModifications() {
+        const modificationsGroupedByObjectIdentifier = _.groupBy(this.modificationsWithoutCacheChanges, (m) => m.objectIdentifier);
 
-        for (const objectIdentifier in createsGroupedByObjectIdentifier) {
-            const currentCreates = createsGroupedByObjectIdentifier[objectIdentifier];
+        for (const objectIdentifier in modificationsGroupedByObjectIdentifier) {
+            const currentModifications = modificationsGroupedByObjectIdentifier[objectIdentifier];
 
-            const targetCollectionName = currentCreates[0].collection;
+            const targetCollectionName = currentModifications[0].collection;
             const targetCollection = await this.collectionProvider.getCollection(targetCollectionName);
 
-            let mergedPayload = { id: objectIdentifier };
+            const lastModification = currentModifications.at(-1)!;
+            if (lastModification.type === DatawalletModificationType.Delete) {
+                await targetCollection.delete({ id: objectIdentifier });
+                this.deletedObjects.push(objectIdentifier);
 
-            for (const create of currentCreates) {
-                mergedPayload = { ...mergedPayload, ...create.payload };
+                continue;
             }
 
-            const newObject = Serializable.fromUnknown(mergedPayload);
+            let resultingObject: any = {};
+            for (const modification of currentModifications) {
+                switch (modification.type) {
+                    case DatawalletModificationType.Create:
+                        resultingObject = { ...resultingObject, ...modification.payload };
+                        break;
+                    case DatawalletModificationType.Update:
+                        resultingObject = { ...resultingObject, ...modification.payload };
+                        break;
```

Comment:
Could be written as

```suggestion
                    case DatawalletModificationType.Create:
                    case DatawalletModificationType.Update:
                        resultingObject = { ...resultingObject, ...modification.payload };
                        break;
```

This way you make it clear that the two are actually intended to be the same and that it is not a copy&paste error.